### PR TITLE
feat: disable animated transitions by default when `prefers-reduced-motion: reduce`

### PR DESCRIPTION
**Pull request by Mysterious-Dev**
Linked to https://github.com/facebook/docusaurus/issues/8506.

### CODE REVIEW

**Comment by Mysterious-Dev** [comment]
Ready for review. For infima, it's what I can done. As far as the mobile version of the TOC is concerned, this is normally done on the Docusaurus side.

**Comment by slorber** [comment]
Thanks, will review it soon.

Was wondering if maybe we could use this CSS var trick to reduce motion everywhere at once?
https://www.silvestar.codes/articles/using-css-variables-for-reduced-motion-on-a-global-scale/

```css
:root {
--ifm-transition-fast: 100ms;
}

@media (prefers-reduced-motion: reduce) {
  :root {
    --ifm-transition-fast: 0ms;
  }
}
```

I didn't try nor study that solution in depth but that's probably worth exploring so that we avoid spreading the mediaqueries everywhere.

**Comment by Mysterious-Dev** [comment]
Hmm, intresting, I will try this ^^.

**Comment by Mysterious-Dev** [comment]
By the way, can I add somes new variables for this ? Because in css, I have different speed for some transitions.

**Comment by slorber** [comment]
I don't know, if this makes sense we can create "design tokens" for speeds 🤪 I don't know how many different anim speeds we have

**Comment by Mysterious-Dev** [comment]
For modified files, I have 2 different speeds for navbar (one at 250ms and the other to 100ms).

And for dropdown, I have a transition without "speed".

File: packages/core/styles/components/navbar.pcss [hunk 1]
```diff
@@ -217,6 +217,10 @@ html[data-theme='dark'],
     width: var(--ifm-navbar-sidebar-width);
     @mixin transition opacity visibility transform, 250ms, ease-in-out;
 
+    @media (prefers-reduced-motion) {
+      @mixin transition visibility, 250ms, ease-in-out;
+    }
+
```
Comment (slorber) [inline packages/core/styles/components/navbar.pcss:223]: maybe we can simply remove the 250ms? Maybe it wouldn't change much if it was 200ms?

File: packages/core/styles/components/dropdown.pcss [hunk 2]
```diff
@@ -57,6 +57,10 @@
     visibility: hidden;
     z-index: var(--ifm-z-index-dropdown);
     @mixin transition opacity transform visibility;
+
+    @media (prefers-reduced-motion) {
+      @mixin transition visibility;
+    }
```
Comment (slorber) [inline packages/core/styles/components/dropdown.pcss:63]: not sure this is needed

apparently we have these styles:

CleanShot 2023-01-20 at 16 18 35@2x

As far as I understand (was there before) we have a mixin that adds the duration automatically

```js
postcssMixins({
      mixins: {
        transition: (mixin, properties, duration, timing) => {
          duration = duration || 'var(--ifm-transition-fast)';
          timing = timing || 'var(--ifm-transition-timing-default)';

          if (properties.includes(' ')) {
            return {
              'transition-property': properties.replace(/\s/g, ', '),
              'transition-duration': duration,
              'transition-timing-function': timing,
            };
          } else {
            return {
              transition: [properties, duration, timing].join(' '),
            };
          }
        },
      },
    }),
```

File: packages/core/styles/components/navbar.pcss [hunk 3]
```diff
@@ -239,6 +243,10 @@ html[data-theme='dark'],
       top: 0;
       visibility: hidden;
       @mixin transition opacity visibility, 100ms, ease-in-out;
+
+      @media (prefers-reduced-motion) {
+        @mixin transition visibility, 100ms, ease-in-out;
+      }
```
Comment (slorber) [inline packages/core/styles/components/navbar.pcss:249]: similarly, why do the backdrop appear before the sidebar?

I would test with the default 200ms anim delay for both and see if it looks good (not sure but worth testing to see if we can simplify)

File: packages/core/styles/components/dropdown.pcss (hunk 2 above)
Comment (Mysterious-Dev) [reply packages/core/styles/components/dropdown.pcss]: Done.

File: packages/core/styles/components/navbar.pcss (hunk 1 above)
Comment (Mysterious-Dev) [reply packages/core/styles/components/navbar.pcss]: Done.

File: packages/core/styles/components/navbar.pcss (hunk 3 above)
Comment (Mysterious-Dev) [reply packages/core/styles/components/navbar.pcss]: I change the speed and delete media query. Don't hesitate to tell me if I need to change anything else.

**Comment by Mysterious-Dev** [comment]
Any news ?

File: packages/core/styles/common/variables.pcss [hunk 4]
```diff
@@ -149,4 +149,10 @@
   --ifm-z-index-overlay: 400;
 }
 
+@media (prefers-reduced-motion: reduce) {
+  :root {
+    --ifm-transition-fast: 0ms;
+  }
+}
```
Comment (slorber) [inline packages/core/styles/common/variables.pcss:156]:
```suggestion
@media (prefers-reduced-motion: reduce) {
  :root {
    --ifm-transition-fast: 0ms;
    --ifm-transition-slow: 0ms;
  }
}
```

Just removing the slow transitions too, although I'm not sure we actually use any 😅  maybe we will someday

**Review by slorber — APPROVED** [review]
LGTM thanks 👍 

The new mobile menu backdrop anim timing doesn't look bad so we can keep it

**Comment by slorber** [comment]
will be integrated in https://github.com/facebook/docusaurus/pull/8674

**Comment by Mysterious-Dev** [comment]
Nice ! However, the animation visible on the video below is managed by Docusaurus (with the `willChange` attribute etc...). I'm not sure how to support the media query 😅.

https://user-images.githubusercontent.com/40738104/219637116-147bc2ce-2a53-4962-aa44-18c9a6e6677e.mp4

**Comment by slorber** [comment]
Yes, I added a few fixes on the Docusaurus side as well to handle that

**Comment by Mysterious-Dev** [comment]
Perfect, thanks ^^.

**Pull request merged**